### Data Analysis with Panda
- Bike Sales
- Michael Hendricks
- July 20

In [1]:
import pandas as pd
import numpy as np

In [5]:
station_df = pd.read_csv('data/station.csv')
trip_df = pd.read_csv('data/trip.csv', error_bad_lines=False)
weather_df = pd.read_csv('data/weather.csv')

In [18]:
trip_df.head(2)

trip_id         starttime          stoptime    bikeid  tripduration  \
0      431  10/13/2014 10:31  10/13/2014 10:48  SEA00298       985.935   
1      432  10/13/2014 10:32  10/13/2014 10:48  SEA00195       926.375   

     from_station_name                                    to_station_name  \
0  2nd Ave & Spring St  Occidental Park / Occidental Ave S & S Washing...   
1  2nd Ave & Spring St  Occidental Park / Occidental Ave S & S Washing...   

  from_station_id to_station_id usertype gender  birthyear  
0          CBD-06         PS-04   Member   Male     1960.0  
1          CBD-06         PS-04   Member   Male     1970.0

In [20]:
trip_df[['tripduration']].mean()

tripduration    1178.295675
dtype: float64

### 1. The average trip duration for a borrowed bicycle is 1178.295675

In [23]:
trip_df['birthyear'].mode()

0    1987.0
dtype: float64

### 2. The most common birth year of a bicycle sharer is 1987, and since that's the year I was born, that makes them 33

In [25]:
3. #weather_df

In [34]:
station_df.head()

station_id                    name        lat        long install_date  \
0      BT-01      3rd Ave & Broad St  47.618418 -122.350964   10/13/2014   
1      BT-03       2nd Ave & Vine St  47.615829 -122.348564   10/13/2014   
2      BT-04  6th Ave & Blanchard St  47.616094 -122.341102   10/13/2014   
3      BT-05  2nd Ave & Blanchard St  47.613110 -122.344208   10/13/2014   
4     CBD-03      7th Ave & Union St  47.610731 -122.332447   10/13/2014   

   install_dockcount modification_date  current_dockcount decommission_date  
0                 18               NaN                 18               NaN  
1                 16               NaN                 16               NaN  
2                 16               NaN                 16               NaN  
3                 14               NaN                 14               NaN  
4                 20               NaN                 20               NaN

In [32]:
#station_df[['name','current_dockcount']]
station_df[['current_dockcount']].mean()

current_dockcount    16.517241
dtype: float64

### 4. The average number of bikes at a given station is 16.517241

In [60]:
# notnull() filters out NaN returns
modified_stations = station_df[station_df['modification_date'].notnull()][['name','install_dockcount', 'modification_date', 'current_dockcount']].set_index('name')

In [88]:
#modified_stations

In [76]:
#modified_stations[['install_dockcount']]
modified_stations[['install_dockcount']].mean()

install_dockcount    17.529412
dtype: float64

In [77]:
#modified_stations[['current_dockcount']]
modified_stations[['current_dockcount']].mean()

current_dockcount    13.764706
dtype: float64

5. When a bike station is modified it is more likely that it's lose bikes because the average dock count of stations before modifications was higher than the average dock count of those same stations after modifications

In [87]:
weather_df

Date  Max_Temperature_F  Mean_Temperature_F  Min_TemperatureF  \
0    10/13/2014                 71                62.0                54   
1    10/14/2014                 63                59.0                55   
2    10/15/2014                 62                58.0                54   
3    10/16/2014                 71                61.0                52   
4    10/17/2014                 64                60.0                57   
..          ...                ...                 ...               ...   
684   8/27/2016                 72                66.0                61   
685   8/28/2016                 75                68.0                59   
686   8/29/2016                 81                68.0                55   
687   8/30/2016                 70                64.0                57   
688   8/31/2016                 71                65.0                59   

     Max_Dew_Point_F  MeanDew_Point_F  Min_Dewpoint_F  Max_Humidity  \
0                 55               51              46            87   
1                 52               51              50            88   
2                 53               50              46            87   
3                 49               46              42            83   
4                 55               51              41            87   
..               ...              ...             ...           ...   
684               57               54              50            81   
685               54               52              50            80   
686               55               53              50            89   
687               55               53              52            83   
688               61               56              52            90   

     Mean_Humidity  Min_Humidity  ...  Mean_Sea_Level_Pressure_In  \
0               68            46  ...                       29.79   
1               78            63  ...                       29.75   
2               77            67  ...                       29.71   
3               61            36  ...                       29.95   
4               72            46  ...                       29.78   
..             ...           ...  ...                         ...   
684             65            46  ...                       29.88   
685             65            44  ...                       30.06   
686             65            39  ...                       30.02   
687             69            53  ...                       30.01   
688             77            63  ...                       29.97   

     Min_Sea_Level_Pressure_In  Max_Visibility_Miles  Mean_Visibility_Miles  \
0                        29.65                    10                     10   
1                        29.54                    10                      9   
2                        29.51                    10                      9   
3                        29.81                    10                     10   
4                        29.73                    10                     10   
..                         ...                   ...                    ...   
684                      29.77                    10                     10   
685                      30.01                    10                     10   
686                      29.95                    10                     10   
687                      29.97                    10                     10   
688                      29.96                    10                     10   

     Min_Visibility_Miles  Max_Wind_Speed_MPH  Mean_Wind_Speed_MPH  \
0                       4                  13                    4   
1                       3                  10                    5   
2                       3                  18                    7   
3                      10                   9                    4   
4                       6                   8                    3   
..                    ...                 ...  

In [106]:
days_with_prec = weather_df[weather_df['Precipitation_In'] > 0][['Date','Precipitation_In', 'Events']]

In [116]:
days_with_prec['Date'] = pd.to_datetime(days_with_prec['Date'], format='%m/%d/%Y')

In [115]:
days_with_prec

Date  Precipitation_In Events
1   2014-10-14              0.11   Rain
2   2014-10-15              0.45   Rain
4   2014-10-17              0.14   Rain
5   2014-10-18              0.31   Rain
7   2014-10-20              0.44   Rain
..         ...               ...    ...
633 2016-07-07              0.06   Rain
634 2016-07-08              0.14   Rain
635 2016-07-09              0.07   Rain
648 2016-07-22              0.21   Rain
664 2016-08-07              0.03   Rain

[294 rows x 3 columns]

In [118]:
days_with_prec.loc[days_with_prec['Date'].dt.month == 12]

Date  Precipitation_In             Events
52  2014-12-04              0.05               Rain
53  2014-12-05              0.09               Rain
54  2014-12-06              0.25               Rain
56  2014-12-08              0.45               Rain
57  2014-12-09              0.42               Rain
58  2014-12-10              0.50               Rain
59  2014-12-11              0.33               Rain
61  2014-12-13              0.01                NaN
65  2014-12-17              0.16               Rain
66  2014-12-18              0.60               Rain
67  2014-12-19              0.13               Rain
68  2014-12-20              0.60               Rain
71  2014-12-23              0.61               Rain
72  2014-12-24              0.12               Rain
75  2014-12-27              0.12               Rain
76  2014-12-28              0.06               Rain
414 2015-12-01              0.39               Rain
415 2015-12-02              0.06               Rain
416 2015-12-03              0.52               Rain
417 2015-12-04              0.12               Rain
418 2015-12-05              0.81               Rain
419 2015-12-06              0.55               Rain
420 2015-12-07              1.06               Rain
421 2015-12-08              1.51               Rain
422 2015-12-09              0.56  Rain-Thunderstorm
423 2015-12-10              0.63  Rain-Thunderstorm
424 2015-12-11              0.01               Rain
425 2015-12-12              0.56               Rain
426 2015-12-13              0.11               Rain
428 2015-12-15              0.02           Fog-Rain
429 2015-12-16              0.13               Rain
430 2015-12-17              0.82               Rain
431 2015-12-18              0.54               Rain
432 2015-12-19              0.01               Rain
433 2015-12-20              0.19               Rain
434 2015-12-21              0.83               Rain
435 2015-12-22              0.12               Rain
436 2015-12-23              0.09               Rain
437 2015-12-24              0.11               Rain
438 2015-12-25              0.05               Rain
440 2015-12-27              0.32               Rain
441 2015-12-28              0.03               Rain